src/ansys/dpf/composites/sampling_point_solid_stack.py

```diff
@@ -0,0 +1,794 @@
+    get_polar_plot_from_sp,
+    get_result_plots_from_sp,
+)
+from .failure_criteria import CombinedFailureCriterion
+from .layup_info._layup_info import (
+    _get_layup_model_context,
+    get_material_names_to_dpf_material_index,
+)
+from .layup_info.material_operators import MaterialOperators
+from .layup_info.material_properties import get_material_metadata
+from .result_definition import FailureMeasureEnum
+from .sampling_point_types import FailureResult, SamplingPoint, SamplingPointFigure
+from .server_helpers import version_older_than
+from .unit_system import get_unit_system
+
+
+class SamplingPointSolidStack(SamplingPoint):
+    """Implements the ``Sampling Point`` object for a stack of solid elements.
+
+    This class provides for plotting the lay-up and results at a certain point of the
+    layered structure. The results, including ``analysis_plies``, ``e1``, ``s12``, and
+    ``failure_modes``, are always from the bottom to the top of the laminate (along
+    the element normal direction). Postprocessing results such as ``e1`` are returned
+    as flat arrays where ``self.spots_per_ply`` can be used to compute the index for
+    a certain ply.
+
+    Parameters
+    ----------
+    name :
+        Name of the object.
+    element_id :
+       Element label of a solid element. The sampling will automatically select to
+       full stack of solid elements where this element is located.
+
+    Notes
+    -----
+    The results of layered elements are stored per integration point. A layered shell element
+    has a number of in-plane integration points (depending on the integration scheme) and
+    typically three integration points through the thickness. The through-the-thickness
+    integration points are called `spots`. They are typically at the ``BOTTOM``
+    and ``TOP`` of the layer. This notation is used here to identify the corresponding data.
+    Note that ``MIDDLE`` is not available for solid elements.
+
+    The ``SamplingPoint`` class returns three results per layer (one for each spot) because
+    the results of the in-plane integration points are interpolated to the centroid of the element.
+    The following table shows an example of a laminate with three layers. So a result, such as
+    ``s1`` has nine values, three for each ply.
+
+    +------------+------------+------------------------+
+    | Layer      | Index      | Spot                   |
+    +============+============+========================+
+    | Layer 3    | - 5        | - TOP of Layer 3       |
+    |            | - 4        | - BOTTOM of Layer 3    |
+    +------------+------------+------------------------+
+    | Layer 2    | - 3        | - TOP of Layer 2       |
+    |            | - 2        | - BOTTOM of Layer 2    |
+    +------------+------------+------------------------+
+    | Layer 1    | - 1        | - TOP of Layer 1       |
+    |            | - 0        | - BOTTOM of Layer 1    |
+    +------------+------------+------------------------+
```

Comment:
Now I'm confused again.. why is the description talking about 9 values, but the table only showing 6?